### PR TITLE
Re-implemented sftp code

### PR DESCRIPTION
**Pull request by jw2476**
I pulled the sftp code out of an older commit, it's not being run in hedge.py yet because I can't think of a good trigger, any ideas.

### CODE REVIEW

**Comment by jw2476** [comment]
Is it possible for the Pi Zero to tell the Pi3 it has a video file waiting, the the Pi3 can sftp it when it has time ? Or is once an hour better ?

File: app/sftp.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import os
+
+
+def pull_videos(ip, username, password):
+    os.chdir('/home/pi/Videos')  # Set pi 3 cwd to Videos
+    remoteCam = pysftp.Connection(ip, username=username, password=password)
+    remoteCam.chdir('/home/pi/Videos')  # Set pi 0 cwd to Videos
+    files = remoteCam.listdir()
+    for file in files:
+        if '.mp4' in file and not '1stPASS' in file:
```
Comment (sebbacon) [inline app/sftp.py]: What is the meaning of `1stPASS`? Maybe add a comment in the code.

File: app/hedge.py [hunk 2]
```diff
@@ -164,8 +166,19 @@ def main():
         time_taken = end_time - start_time
         if time_taken < cycle_time:
             time.sleep(cycle_time - time_taken)
+        return None
+
+    elif last_ran != datetime.now().strftime('%H'):
+        print(str(last_ran) + '          ' + datetime.now().strftime('%H'))
+        last_ran = datetime.now().strftime('%H')
+        sftp.pull_videos('10.170.1.11', 'pi', 'hog1hog1')
```
Comment (sebbacon) [inline app/hedge.py]: I would set the ip address, username, and password, as constants at the top of this file

i.e. 
```
PIZERO_IP = '10.179.1.11'
PIZERO_FTP_USERNAME = `pi`
PIZERO_FTP_PASSWORD = `hog1hog1`
```

In terms of security, we should not really have passwords committed in code at all. If we are confident that our pizero is never accessible from the internet, we could either (a) add a comment to that effect next to the password definition, or (b) set up the FTP server to allow anonymous connections.

File: app/sftp.py (hunk 1 above)
Comment (jw2476) [reply app/sftp.py]: 1stPASS is a intermediate file made during video capture

File: app/hedge.py (hunk 2 above)
Comment (jw2476) [reply app/hedge.py]: I agree, maybe use system variables or something

File: app/hedge.py [hunk 3]
```diff
@@ -115,7 +118,7 @@ def cleanup():
 #  =======================================
 # Main Loop
 #  =======================================
-def main(last_ran, box_id, cycle_time):
+def main(last_ran, box_id, cycle_time, PIZERO_IP, PIZERO_FTP_USERNAME, PIZERO_FTP_PASSWORD):
```
Comment (sebbacon) [inline app/hedge.py]: You don't need to pass these constants around between functions as they are defined at the module scope.

Comment (jw2476) [reply app/hedge.py]: Yeah that's what I thought but I got a load of issues with it, I think it might be because we're using python3.5

**Comment by jw2476** [comment]
Hey @sebbacon I can't get this to merge could you have a look please

**Comment by sebbacon** [comment]
@jw2476 I've manually resolved all the conflicts as far as I can, but obviously I'm not familiar with the underlying logic, so a double check of how I've done it in the last commit would be a good idea!

**Pull request merged**